packages/compass-collection/src/modules/collection-tab.ts

```diff
@@ -22,32 +31,82 @@ type CollectionThunkAction<R, A extends AnyAction = AnyAction> = ThunkAction<
     dataService: DataService;
     workspaces: ReturnType<typeof workspacesServiceLocator>;
     experimentationServices: ReturnType<typeof experimentationServiceLocator>;
+    logger: Logger;
+    preferences: PreferencesAccess;
   },
   A
 >;
 
+export enum SchemaAnalysisStatus {
+  INITIAL = 'initial',
+  ANALYZING = 'analyzing',
+  COMPLETED = 'completed',
+  ERROR = 'error',
+}
+
+type SchemaAnalysis = {
```

Comment:
`| null` in various fields is a sign that we can apply discriminated fields to improve.

You can break this into a type for each Status

```typescript

type SchemaAnalysis = SchemaAnalysisError | SchemaAnalysisInitial | SchemaAnalysis | SchemaAnalysisCompleted`

```

each type shares the `status` but the `status` helps the type checker see what fields exist based on the status

e.g., an `error` status only exists for `SchemaAnalysisError`

The TypeScript handbook has an example [here](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions)

Reply:
This is very helpful! I'm gonna move these types to their own file as well!